SYCL/ESIMD/api/functional/functions/functions_atomic_update_inc_dec.cpp

```diff
@@ -0,0 +1,250 @@
+    // Collect the indexess that has been changed.
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      if (filter(i) == 1) {
+        changed_elems_indexes.push_back(offsets[i]);
+      }
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    changed_elems_indexes.push_back(std::numeric_limits<size_t>::max());
+    auto updated_elem_next_index = changed_elems_indexes.begin();
+
+    const DataT &expected = base_value;
+    DataT expected_after_change =
+        get_expected_value<ChosenOperator>(base_value, NumberIteractions);
+    // Verify that values, that do not was changed has initial values.
+    for (size_t i = 0; i < NumElems; ++i) {
+      // If current index is less than changed element index verify that this
+      // element hasn't been changed.
+      const DataT &retrieved = shared_init_values[i];
+      if (i < *updated_elem_next_index) {
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type,
+                             "value that should not be updated", operator_type);
+        }
+      } else {
+        if (expected_after_change != retrieved) {
+          passed = fail_test(i, expected_after_change, retrieved, data_type,
+                             "value that should be updated", operator_type);
+        }
+        updated_elem_next_index++;
+      }
+    }
+
+    return passed;
+  }
+
+private:
+  bool fail_test(size_t i, DataT expected, DataT retrieved,
+                 const std::string &data_type,
+                 const std::string &verification_type,
+                 const std::string &operator_type) {
+    log::fail(TestDescriptionT(data_type), "Unexpected value at index ", i,
+              ", retrieved: ", retrieved, ", expected: ", expected,
+              ", for verification type: ", verification_type,
+              ", with number elements to change: ", NumElemsToChange,
+              ", with operator type: ", operator_type);
+
+    return false;
+  }
+};
+
+int main(int, char **) {
+  sycl::queue queue(esimd_test::ESIMDSelector{},
+                    esimd_test::createExceptionHandler());
+
+  bool passed = true;
+
+  const auto uint_types = named_type_pack<uint32_t>::generate("uint32_t");
+  const auto single_size = get_sizes<16>();
+  const auto nd_range_dims = integer_pack<1, 2, 3>::generate_unnamed();
+  const auto num_elems_to_change = integer_pack<1, 8>::generate_unnamed();
+
+  const auto filter_types =
+      unnamed_type_pack<functions::masks::ChangeByStep,
+                        functions::masks::ChangeAll,
+                        functions::masks::ChangeNothing>::generate();
+  const auto atomic_op_types =
+      value_pack<atomic_op, atomic_op::inc, atomic_op::dec>::generate_named(
+          " atomic_op::inc", "atomic_op::dec");
+  const auto algorithm_to_change_types =
+      value_pack<functions::offset, functions::offset::all,
```

Comment:
```suggestion
  const auto algorithm_to_change_offset =
      value_pack<functions::offset, functions::offset::all,
```
Did you mean this?

Reply:
yes, applied this comment [here](https://github.com/intel/llvm-test-suite/pull/939/commits/3a3e37f3d8b5c247a3e92f48f51646d4bf27f524#diff-afaea17121d6706af52229dcfb462795dfe86fde5c98f09b836505a741830b6bR228-R231)